contracts/crosschain/draft-ERC7786Receiver.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {IERC7786GatewayDestinationPassive, IERC7786Receiver} from "../interfaces/draft-IERC7786.sol";
+
+/**
+ * @dev Base implementation of an ERC-7786 compliant cross-chain message receiver.
+ *
+ * This abstract contract exposes the `receiveMessage` function that is used in both active and passive mode for
+ * communication with (on or multiple) destination gateways. This contract leaves two function unimplemented:
+ *
+ * `_isKnownGateway(address)`, an internal getter used to verify weither an address is recognised by the contract as a
+ * valid ERC-7786 destination gateway. One or multiple gateway can be supported. Note that any malicious address for
+ * which this function returns true would be able to impersonnate any account on any other chain sending any message.
+ *
+ * `_processMessage`, the internal function that will be called with any message that has been validated.
+ */
+abstract contract ERC7786Receiver is IERC7786Receiver {
+    error ERC7786ReceiverInvalidGateway(address gateway);
+
+    /// @inheritdoc IERC7786Receiver
+    function receiveMessage(
+        address gateway,
+        bytes calldata gatewayMessageKey,
+        string calldata source,
+        string calldata sender,
+        bytes calldata payload,
+        bytes[] calldata attributes
+    ) public payable virtual {
+        if (_isKnownGateway(msg.sender)) {
+            // Active mode
+            // no extra check
```

Comment:
shouldn't this check that `gatewayMessageKey` and `gateway` are zero https://github.com/ethereum/ERCs/pull/673/files#diff-07848a22c16b11582927b37d76b2c54a0617b1444b30d11bca783ac98eaed01fR150? or is that checked by the gateway?

Reply:
One answer is that any value should come with a corresponding attribute, and that is how it would be visible from the `processMessage` (instead of using msg.value)

Reply:
@frangio

Reply:
Yeah I specified it as "SHOULD" so this implementation is compliant. But it's worth considering if it should be "MUST". I don't see any concrete issue with allowing any values and ignoring them.